python/cudf/cudf/utils/utils.py

```diff
@@ -159,8 +159,9 @@ def _external_only_api(func, alternative=""):
     @functools.wraps(func)
     def wrapper(*args, **kwargs):
         # Check the immediately preceding frame to see if it's in cudf.
-        frame, lineno = next(traceback.walk_stack(None))
-        fn = frame.f_code.co_filename
+        pre_frame = traceback.extract_stack(limit=2)[0]
+        fn = pre_frame.filename
+        lineno = pre_frame.lineno
```

Comment:
What has changed such that this is necessary?

Reply:
I am not sure, but I think it is because we now use a context always. 

For some reason, [`walk_stack()`](https://github.com/python/cpython/blob/main/Lib/traceback.py#L383) and [`extract_stack()`](https://github.com/python/cpython/blob/main/Lib/traceback.py#L383) use different default start frames.